Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -133,13 +133,17 @@ public struct PackageBuilder {
     /// The path of the package.
     private let packagePath: AbsolutePath
 
+    /// The filesystem package builder will run on.
+    private let fileSystem: FileSystem
+
     /// Create a builder for the given manifest and package `path`.
     ///
     /// - Parameters:
     ///   - path: The root path of the package.
-    public init(manifest: Manifest, path: AbsolutePath) {
+    public init(manifest: Manifest, path: AbsolutePath, fileSystem: FileSystem = localFileSystem) {
         self.manifest = manifest
         self.packagePath = path
+        self.fileSystem = fileSystem
```

Comment:
One thing I wanted to do as part of this was to rewrite the logic to use a FileSystem rooted at `packagePath` (using `RerootedFileSystem`, when dealing with a local FS)... do you think that makes sense, and is it best as a separate commit?